pallets/tendermint-light-client/src/types.rs

```diff
@@ -0,0 +1,1101 @@
+
+    fn try_from(value: tendermint::Hash) -> Result<Self, Self::Error> {
+        match value {
+            TendermintHash::Sha256(bytes) => {
+                if bytes.len() == hash::SHA256_HASH_SIZE {
+                    Ok(TendermintHashStorage::Some(H256::from_slice(&bytes)))
+                } else {
+                    Err(ConversionError::WithContext("Invalid hash size"))
+                }
+            }
+            TendermintHash::None => Ok(TendermintHashStorage::None),
+        }
+    }
+}
+
+impl From<TendermintHashStorage> for TendermintHash {
+    fn from(value: TendermintHashStorage) -> Self {
+        match value {
+            TendermintHashStorage::Some(hash) => TendermintHash::Sha256(hash.0),
+            TendermintHashStorage::None => TendermintHash::None,
+        }
+    }
+}
+
+#[derive(Encode, Decode, Clone, RuntimeDebug, TypeInfo, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct TrustThresholdStorage {
+    pub numerator: u64,
+    pub denominator: u64,
+}
+
+impl TrustThresholdStorage {
+    pub fn new(numerator: u64, denominator: u64) -> Self {
+        Self {
+            numerator,
+            denominator,
+        }
+    }
+}
+
+impl TryFrom<TrustThresholdStorage> for TrustThreshold {
+    type Error = ConversionError;
+
+    fn try_from(value: TrustThresholdStorage) -> Result<Self, Self::Error> {
+        Self::new(value.numerator, value.denominator).map_err(ConversionError::Upstream)
+    }
+}
+
+#[derive(Encode, Decode, Clone, RuntimeDebug, TypeInfo, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct LightClientOptionsStorage {
+    /// Defines what fraction of the total voting power of a known
+    /// and trusted validator set is sufficient for a commit to be
+    /// accepted going forward.
+    pub trust_threshold: TrustThresholdStorage,
+    /// How long a validator set is trusted for (must be shorter than the chain's
+    /// unbonding period) in secs
+    pub trusting_period: u64,
+    /// Correction parameter dealing with only approximately synchronized clocks.
+    /// The local clock should always be ahead of timestamps from the blockchain; this
+    /// is the maximum amount that the local clock may drift behind a timestamp from the
+    /// blockchain.
+    pub clock_drift: u64,
+}
+
+impl LightClientOptionsStorage {
+    pub fn new(
+        trust_threshold: TrustThresholdStorage,
+        trusting_period: u64,
+        clock_drift: u64,
+    ) -> Self {
+        Self {
+            trust_threshold,
+            trusting_period,
+            clock_drift,
+        }
+    }
+}
+
+impl Default for LightClientOptionsStorage {
+    fn default() -> Self {
+        Self {
+            trust_threshold: TrustThresholdStorage {
+                numerator: 1,
+                denominator: 3,
+            },
+            trusting_period: 1210000, // 2 weeks
+            clock_drift: 5,
+        }
+    }
+}
```

Comment:
What do we need the defaults for? Testing? If so, can we mark this with a proper `#[cfg(test)]` annotation?

Reply:
done